### ISSUE TITLE
ProcessStatisticsResource.mem_quota: Numeric value (2147483648) out of range of int

### ISSUE THREAD

**Issue by 4prohit** [cloudfoundry/cf-java-client#952]
I am trying to retrieve statistics of all the instances for the applications running in PCF.

I am using `CF Java Client` with latest version `3.15.0.RELEASE` . I am using below dependencies in my project.

**pom.xml**
```
<dependency>
	<groupId>org.cloudfoundry</groupId>
	<artifactId>cloudfoundry-client-reactor</artifactId>
	<version>3.15.0.RELEASE</version>
</dependency>
<dependency>
	<groupId>org.cloudfoundry</groupId>
	<artifactId>cloudfoundry-operations</artifactId>
	<version>3.15.0.RELEASE</version>
</dependency>
```

I have below code block which retrieves the Organisation and Space details using `CloudFoundryOperations` and then uses `CloudFoundryClient` to retrieve application and instance details.

**CFClient.java**:
```
public void run(ApplicationArguments args) throws Exception {
        final OrganizationInfoRequest organizationInfoRequest = OrganizationInfoRequest.builder()
                .name(externalConfig.getOrganization())
                .build();
        final Mono<OrganizationDetail> organizationDetailMono = cloudFoundryOperations.organizations().get(organizationInfoRequest);
        final String organizationId = organizationDetailMono.map(OrganizationDetail::getId).block();
        logger.info("Organization Name: {}, Id: {}", externalConfig.getOrganization(), organizationId);

        final GetSpaceRequest getSpaceRequest = GetSpaceRequest.builder()
                .name(externalConfig.getSpace())
                .build();
        final Mono<SpaceDetail> spaceDetailMono = cloudFoundryOperations.spaces().get(getSpaceRequest);
        final String spaceId = spaceDetailMono.map(SpaceDetail::getId).block();
        logger.info("Space Name: {}, Id: {}", externalConfig.getSpace(), spaceId);

        logger.info("Retrieving information for applications: {}", externalConfig.getApplications());
        final ListApplicationsRequest listApplicationsRequest = ListApplicationsRequest.builder()
                .organizationId(organizationId)
                .spaceId(spaceId)
                .names(externalConfig.getApplications())
                .build();
        final Mono<ListApplicationsResponse> listApplicationsResponseMono = cloudFoundryClient.applicationsV3().list(listApplicationsRequest);
        final List<ApplicationResource> applicationResources = listApplicationsResponseMono.map(ListApplicationsResponse::getResources).block();
        for (ApplicationResource applicationResource : applicationResources) {
            final ListApplicationProcessesRequest listApplicationProcessesRequest = ListApplicationProcessesRequest.builder()
                    .applicationId(applicationResource.getId())
                    .build();
            final Mono<ListApplicationProcessesResponse> listApplicationProcessesResponseMono = cloudFoundryClient.applicationsV3().listProcesses(listApplicationProcessesRequest);
            final List<ProcessResource> processResources = listApplicationProcessesResponseMono.map(ListApplicationProcessesResponse::getResources).block();
            for (ProcessResource processResource : processResources) {
                logger.info("Application Name: {}, Id: {}, State: {}, Process Id: {}, Instances: {}", applicationResource.getName(), applicationResource.getId(), applicationResource.getState(), processResource.getId(), processResource.getInstances());
                final GetProcessStatisticsRequest getProcessStatisticsRequest = GetProcessStatisticsRequest.builder()
                        .processId(processResource.getId())
                        .build();
                final Mono<GetProcessStatisticsResponse> getProcessStatisticsResponseMono = cloudFoundryClient.processes().getStatistics(getProcessStatisticsRequest);
                final List<ProcessStatisticsResource> processStatisticsResources = getProcessStatisticsResponseMono.map(GetProcessStatisticsResponse::getResources).block();
                for (ProcessStatisticsResource processStatisticsResource : processStatisticsResources) {
                    logger.info("Instance Index: {}, State: {}", processStatisticsResource.getIndex(), processStatisticsResource.getState());
                }
            }
        }
    }
```

But I am not able to read the `ProcessStatisticsResource` due to Numeric value (2147483648) is out of range of int. I can see that `ProcessStatisticsResource` has `memoryQuota` and `diskQuota` of `Integer` type.

**ProcessStatisticsResource.class**
```
public final class ProcessStatisticsResource extends _ProcessStatisticsResource {
    @Nullable
    private final String details;
    private final Integer diskQuota;
    private final Integer fileDescriptorQuota;
    private final String host;
    private final Integer index;
    private final List<PortMapping> instancePorts;
    @Nullable
    private final String isolationSegment;
    private final Integer memoryQuota;
    @Nullable
    private final ProcessState state;
    private final String type;
    private final Integer uptime;
    private final ProcessUsage usage;
    .
    .
    .
```

**Stacktrace**:
```
Caused by: org.cloudfoundry.reactor.util.JsonParsingException: Numeric value (2147483648) out of range of int
 at [Source: (byte[])"{
  "resources": [
    {
      "type": "web",
      "index": 0,
      "state": "RUNNING",
      "usage": {
        "time": "2019-03-07T03:35:10+00:00",
        "cpu": 0.0,
        "mem": 396468224,
        "disk": 456839168
      },
      "host": "10.92.205.52",
      "uptime": 245924,
      "mem_quota": 2147483648,
      "disk_quota": 1073741824,
      "fds_quota": 16384,
      "isolation_segment": null,
      "instance_ports": [
        {
          "external": 61001,
          "internal": 8080"[truncated 30 bytes]; line: 15, column: 20] (through reference chain: org.cloudfoundry.client.v3.processes.GetProcessStatisticsResponse$Json["resources"]->java.util.ArrayList[0]->org.cloudfoundry.client.v3.processes.ProcessStatisticsResource$Json["mem_quota"])
	at org.cloudfoundry.reactor.util.JsonCodec.lambda$null$0(JsonCodec.java:51) ~[cloudfoundry-client-reactor-3.15.0.RELEASE.jar:na]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:100) ~[reactor-core-3.2.6.RELEASE.jar:3.2.6.RELEASE]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:114) ~[reactor-core-3.2.6.RELEASE.jar:3.2.6.RELEASE]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:114) ~[reactor-core-3.2.6.RELEASE.jar:3.2.6.RELEASE]
	at reactor.ipc.netty.channel.FluxReceive.drainReceiver(FluxReceive.java:213) ~[reactor-netty-0.7.9.RELEASE.jar:0.7.9.RELEASE]
	at reactor.ipc.netty.channel.FluxReceive.onInboundNext(FluxReceive.java:329) ~[reactor-netty-0.7.9.RELEASE.jar:0.7.9.RELEASE]
	at reactor.ipc.netty.channel.ChannelOperations.onInboundNext(ChannelOperations.java:311) ~[reactor-netty-0.7.9.RELEASE.jar:0.7.9.RELEASE]
	at reactor.ipc.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:628) ~[reactor-netty-0.7.9.RELEASE.jar:0.7.9.RELEASE]
	at reactor.ipc.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:138) ~[reactor-netty-0.7.9.RELEASE.jar:0.7.9.RELEASE]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) ~[netty-codec-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297) ~[netty-codec-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at reactor.ipc.netty.http.HttpOperations.lambda$static$2(HttpOperations.java:280) ~[reactor-netty-0.7.9.RELEASE.jar:0.7.9.RELEASE]
	at reactor.ipc.netty.ReactorNetty$ExtractorHandler.channelRead(ReactorNetty.java:366) ~[reactor-netty-0.7.9.RELEASE.jar:0.7.9.RELEASE]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102) ~[netty-codec-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:438) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) ~[netty-codec-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297) ~[netty-codec-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:253) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1436) ~[netty-handler-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1203) ~[netty-handler-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1247) ~[netty-handler-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502) ~[netty-codec-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441) ~[netty-codec-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278) ~[netty-codec-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:677) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:612) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:529) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:491) ~[netty-transport-4.1.33.Final.jar:4.1.33.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:905) ~[netty-common-4.1.33.Final.jar:4.1.33.Final]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_92]
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Assembly trace from producer [reactor.core.publisher.MonoPeek] :
	reactor.core.publisher.Mono.checkpoint(Mono.java:1681)
	org.cloudfoundry.reactor.client.v3.processes.ReactorProcesses.getStatistics(ReactorProcesses.java:61)
Error has been observed by the following operator(s):
	|_	Mono.checkpoint ⇢ org.cloudfoundry.reactor.client.v3.processes.ReactorProcesses.getStatistics(ReactorProcesses.java:61)
```
```
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Numeric value (2147483648) out of range of int
 at [Source: (byte[])"{
  "resources": [
    {
      "type": "web",
      "index": 0,
      "state": "RUNNING",
      "usage": {
        "time": "2019-03-07T03:35:10+00:00",
        "cpu": 0.0,
        "mem": 396468224,
        "disk": 456839168
      },
      "host": "10.92.205.52",
      "uptime": 245924,
      "mem_quota": 2147483648,
      "disk_quota": 1073741824,
      "fds_quota": 16384,
      "isolation_segment": null,
      "instance_ports": [
        {
          "external": 61001,
          "internal": 8080"[truncated 30 bytes]; line: 15, column: 20] (through reference chain: org.cloudfoundry.client.v3.processes.GetProcessStatisticsResponse$Json["resources"]->java.util.ArrayList[0]->org.cloudfoundry.client.v3.processes.ProcessStatisticsResource$Json["mem_quota"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:394) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:353) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1711) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:290) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:189) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:161) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1284) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:189) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:161) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1284) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3091) ~[jackson-databind-2.9.8.jar:2.9.8]
	at org.cloudfoundry.reactor.util.JsonCodec.lambda$null$0(JsonCodec.java:49) ~[cloudfoundry-client-reactor-3.15.0.RELEASE.jar:na]
	... 52 common frames omitted
Caused by: com.fasterxml.jackson.core.JsonParseException: Numeric value (2147483648) out of range of int
 at [Source: (byte[])"{
  "resources": [
    {
      "type": "web",
      "index": 0,
      "state": "RUNNING",
      "usage": {
        "time": "2019-03-07T03:35:10+00:00",
        "cpu": 0.0,
        "mem": 396468224,
        "disk": 456839168
      },
      "host": "10.92.205.52",
      "uptime": 245924,
      "mem_quota": 2147483648,
      "disk_quota": 1073741824,
      "fds_quota": 16384,
      "isolation_segment": null,
      "instance_ports": [
        {
          "external": 61001,
          "internal": 8080"[truncated 30 bytes]; line: 15, column: 30]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1804) ~[jackson-core-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:693) ~[jackson-core-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.core.base.ParserBase.convertNumberToInt(ParserBase.java:886) ~[jackson-core-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.core.base.ParserBase._parseIntValue(ParserBase.java:801) ~[jackson-core-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.core.base.ParserBase.getIntValue(ParserBase.java:645) ~[jackson-core-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:472) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:452) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127) ~[jackson-databind-2.9.8.jar:2.9.8]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288) ~[jackson-databind-2.9.8.jar:2.9.8]
	... 70 common frames omitted
```

I think the datatype for `memoryQuota` and `diskQuota` should be `Double`.

**Comment by twoseat** [comment]
Can you tell me what the output of `cf quotas` is on the command line? (Sanitise if there's anything confidential in the output) I believe the default unit for `mem_quota` and `disk_quota` is MB, so you appear to have a memory quota of around 2 petabytes, which is generous.

**Comment by rohit4patil** [comment]
Hi @twoseat ,
Please find below the quota details:
```
Total Memory           100G
Instance Memory        unlimited
Routes                 1000
Services               100
Paid service plans     allowed
App instance limit     unlimited
Reserved Route Ports   0
```

**Comment by keviningvalson** [comment]
2019-03-13T10:39:31.400-05:00 [APP/PROC/WEB/0] [OUT] "exter"[truncated 146 bytes]; line: 15, column: 20] (through reference chain: org.cloudfoundry.client.v3.processes.GetProcessStatisticsResponse$Json["resources"]->java.util.ArrayList[0]->org.cloudfoundry.client.v3.processes.ProcessStatisticsResource$Json["mem_quota"])


com.fasterxml.jackson.databind.JsonMappingException: Numeric value (3221225472) out of range of int

**Comment by twoseat** [comment]
Thanks for the additional info @rohit4patil and @keviningvalson - I agree that in theory changing the field type would 'fix' the problem, but I'm trying to understand where the problem comes from as we didn't pick `Integer` by chance :)

If I run `cf create-quota test -m 2147483648M` I get a `Server error, status code: 500, error code: 10011, message: Database error`, which I believe is because the corresponding database field is also an `Integer` (if I decrease the request by 1 it works fine). Similarly the [cloud controller](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/controllers/runtime/quota_definitions_controller.rb) seems to define these fields as `Integer`s. Yet I see as you do responses such as `"mem_quota": 3221225472`. I'm trying to find out why this is.

**Comment by twoseat** [comment]
I heard back from the relevant team and "In mysql at least, it is a `int(11)`". That tells me that something else in the chain is modifying the value incorrectly, but the java client shouldn't be changing to accomodate that. I've not given up trying to find where that rogue change is being made, as clearly it does cause problems for us (and hence you).

**Comment by dmikusa-pivotal** [comment]
There were similar issues fixed in 4.16.0.RELEASE and 5.6.0.RELEASE. I think it will resolve the issue reported here, although 3.x is no longer supported so patched did not make it into that version.

I think this boils down to a specific issue with process stats. The process stats are not stored in the database, they come out of Loggregator from the container metrics. They are reported in bytes (CAPI stores things in MB), so an app with more than 2G of memory can overflow the integer.
